packages/react-ui/src/utils/storybookNavigation.ts

```
@@ -83,26 +101,118 @@ export class StorybookNavigator {
-      iframe.src = iframe.src; // Force reload
+      try {
+        // Only reload if the iframe appears to be on a different story
+        const currentPath = new URLSearchParams(window.location.search).get('path');
+        if (currentPath && !iframe.src.includes(encodeURIComponent(currentPath))) {
+          // Build the iframe URL
+          const iframeBaseUrl = iframe.src.split('?')[0];
+          iframe.src = `${iframeBaseUrl}?path=${currentPath}&viewMode=story`;
+        }
+      } catch (error) {
+        // Fallback to simple reload
+        iframe.src = iframe.src;
```

Biome (2.1.2)

[error] src is assigned to itself.

This is where is assigned.

Self assignments have no effect and can be removed.

(lint/correctness/noSelfAssign)